### ISSUE TITLE
AssertionError: No Entity registered with name <class>

### ISSUE THREAD

**Issue by advishnuprasad** [proteanhq/protean#116]
When I use reference as a class name instead of a string,

I get

```AssertionError: No Entity registered with name <class 'Account'>```

Example:
`account = field.Reference(Account)` instead of `account = field.Reference("Account")`

It works on DictModel. When I use SqlAlchemyModel it failed.

**Comment by subhashb** [comment]
This issue could happen if `Account` Entity is not imported yet. It will be resolved properly by the time the actual attribute call happens, like `some_entity.account`.

My guess is when you initialize it with Dict Repository, `Account` was imported before your entity, while with SQLAlchemy Repository, it was not. Can you test this hypothesis?

**Comment by advishnuprasad** [comment]
It could be. But I do have an import statement in the reference entity file. 

This is the stack trace. 

```
Traceback (most recent call last):
  File "/Users/advishnuprasad/.pyenv/versions/3.6.5/envs/vfc-api/lib/python3.6/site-packages/protean/core/repository/factory.py", line 115, in get_entity
    return self._entity_registry[entity_name]
KeyError: <class 'vfc.account.entities.Account'>

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/advishnuprasad/.pyenv/versions/3.6.5/envs/vfc-api/bin/vfc", line 11, in <module>
    load_entry_point('vfc-api', 'console_scripts', 'vfc')()
  File "/Users/advishnuprasad/.pyenv/versions/3.6.5/envs/vfc-api/lib/python3.6/site-packages/pkg_resources/__init__.py", line 489, in load_entry_point
    return get_distribution(dist).load_entry_point(group, name)
  File "/Users/advishnuprasad/.pyenv/versions/3.6.5/envs/vfc-api/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2843, in load_entry_point
    return ep.load()
  File "/Users/advishnuprasad/.pyenv/versions/3.6.5/envs/vfc-api/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2434, in load
    return self.resolve()
  File "/Users/advishnuprasad/.pyenv/versions/3.6.5/envs/vfc-api/lib/python3.6/site-packages/pkg_resources/__init__.py", line 2440, in resolve
    module = __import__(self.module_name, fromlist=['__name__'], level=0)
  File "/Users/advishnuprasad/team8/vfc/api/src/api/__init__.py", line 15, in <module>
    from vfc.competition.schemas import CompetitionModel, PointModel, \
  File "/Users/advishnuprasad/team8/vfc/vfc/src/vfc/competition/schemas.py", line 16, in <module>
    class EntryModel(SqlalchemyModel):
  File "/Users/advishnuprasad/.pyenv/versions/3.6.5/envs/vfc-api/lib/python3.6/site-packages/protean_sqlalchemy/sa.py", line 40, in __init__
    related_ent = repo_factory.get_entity(field_obj.to_cls)
  File "/Users/advishnuprasad/.pyenv/versions/3.6.5/envs/vfc-api/lib/python3.6/site-packages/protean/core/repository/factory.py", line 117, in get_entity
    raise AssertionError(f'No Entity registered with name {entity_name}')
AssertionError: No Entity registered with name <class 'vfc.account.entities.Account'>
```

**Comment by subhashb** [comment]
Should be resolved by #248
